lxd/resources/pci.go

```diff
@@ -0,0 +1,106 @@
+package resources
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/jaypipes/pcidb"
+	"github.com/pkg/errors"
+
+	"github.com/lxc/lxd/shared/api"
+)
+
+// GetPCI returns a filled api.ResourcesPCI struct ready for use by LXD
+func GetPCI() (*api.ResourcesPCI, error) {
+	pci := api.ResourcesPCI{}
+
+	// Load PCI database
+	pciDB, err := pcidb.New()
+	if err != nil {
+		pciDB = nil
+	}
+
+	// List all PCi devices
```

Comment:
This needs to be under a `sysfsExists` check, same as what's done in `network.go`.
Not all architectures have a PCI bus.